File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.java
class Solution {
    public int findTheWinner(int n, int k) {
        Queue<Integer> queue = new LinkedList<>();
        for(int i=1;i<=n;i++){
            queue.add(i);
        }
        // System.out.println(queue);
        // System.out.println("------");
        while(queue.size()>1){
            int delete = k-1;
            System.out.println(queue);
            while(delete>0){
                queue.add(queue.remove());
                delete--;
            }
            // System.out.println(queue);
            
            queue.remove();
        }
        
        return queue.remove();
    }
}

File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.py
class Solution(object):
    def findTheWinner(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: int
        """
        l=[i for i in range(1,n+1)]
        while len(l)>1:
            remove=k-1
            while remove>0:
                l.append(l.pop(0))
                remove-=1
            l.pop(0)
        return l.pop(0)
        